File: 240418/밭의 높이를 고르게하기/equalizing-the-height-of-the-field.java
import java.util.*;
public class Main {
    public static int N, H, T;
    public static int[] arr = new int[101];
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        N = sc.nextInt();
        H = sc.nextInt();
        T = sc.nextInt();

        for (int i = 0; i < N; i++) 
            arr[i] = Math.abs(H - sc.nextInt());

        int ans = Integer.MAX_VALUE;
        for (int i = 0; i <= N-T; i++){
            int num = 0;
            for (int j = i; j < i+T; j++){
                num += arr[j];
            }
            ans = Math.min(ans, num);
        }
        System.out.println(ans);
    }
}

File: 240612/밭의 높이를 고르게하기/equalizing-the-height-of-the-field.py
N, H, T = map(int, input().split())
arr = list(map(int, input().split()))

board = [0] * N
for i in range(N):
    board[i] = abs(H- arr[i])

ans = int(1e9)
for i in range(0,N- T +1):
    ans = min(ans, sum(board[i: i+T]))
print(ans)